File: scala/sdktest/BatchMessageEntityTest.scala
// Generated basic-flow test for the batch_message entity (model-driven;
// mirrors the java TestEntity generator). A dependency-free scala-cli test
// object driven by SdkEntityTestMain. Runs against the in-memory test
// transport seeded with the shipped BatchMessageTestData.json fixtures.

import java.util.{ArrayList, LinkedHashMap, List => JList, Map => JMap}

import voxgig.thesmsworkssdk.core.{Helpers, ThesmsworksSDK}
import voxgig.thesmsworkssdk.utility.struct.Struct

object BatchMessageEntityTest {

  def run(rep: SdkTestReport): Unit = {
    rep.scope("batch_message.instance") {
      val testsdk = ThesmsworksSDK.testSDK()
      val ent = testsdk.batchMessage(null)
      rep.check("batch_message.instance", ent != null, "expected non-null batch_message entity")
    }

    rep.scope("batch_message.basic") {
      val entityData = Helpers.toMapAny(SdkTestSupport.readJson(
          "../.sdk/test/entity/batch_message/BatchMessageTestData.json"))
      val options = new LinkedHashMap[String, Object]()
      options.put("entity", entityData.get("existing"))
      val client = ThesmsworksSDK.testSDK(options, null)

      val idmap = new LinkedHashMap[String, Object]()
      idmap.put("batch_message01", "BATCH_MESSAGE01")
      idmap.put("batch_message02", "BATCH_MESSAGE02")
      idmap.put("batch_message03", "BATCH_MESSAGE03")
      idmap.put("schedule01", "SCHEDULE01")
      idmap.put("schedule02", "SCHEDULE02")
      idmap.put("schedule03", "SCHEDULE03")
      val now = System.currentTimeMillis()

      // CREATE
      val batchMessageRef01Ent = client.batchMessage(null)
      var batchMessageRef01Data = Helpers.toMapAny(Struct.getprop(
          Struct.getpath(entityData, "new.batch_message"), "batch_message_ref01"))
      val batchMessageRef01DataResult = batchMessageRef01Ent.create(batchMessageRef01Data, null)
      batchMessageRef01Data = Helpers.toMapAny(batchMessageRef01DataResult)
      rep.check("batch_message.create.map", batchMessageRef01Data != null, "expected create result to be a map")

      // REMOVE
      val batchMessageRef01MatchRm0 = new LinkedHashMap[String, Object]()
      batchMessageRef01MatchRm0.put("id", batchMessageRef01Data.get("id"))
      batchMessageRef01Ent.remove(batchMessageRef01MatchRm0, null)
    }
  }
}

File: elixir/test/batch_message_entity_test.exs
# BatchMessage entity test (offline, mock transport)

defmodule Thesmsworks.BatchMessageEntityTest do
  use ExUnit.Case

  alias Voxgig.Struct, as: S
  alias Thesmsworks.Helpers, as: H
  alias Thesmsworks.Json

  defp fixture do
    Json.parse(File.read!("../.sdk/test/entity/batch_message/BatchMessageTestData.json"))
  end

  defp mk_sdk do
    existing = H.or_(S.getpath(fixture(), "existing"), S.jm([]))
    Thesmsworks.test(S.jm(["entity", existing]))
  end

  defp first_id do
    existing = H.or_(S.getpath(fixture(), "existing.batch_message"), S.jm([]))
    keys = S.keysof(existing)
    if keys == [], do: nil, else: hd(keys)
  end

  test "should create instance" do
    sdk = Thesmsworks.test()
    ent = Thesmsworks.batch_message(sdk)
    assert ent != nil
  end

  test "should create then read back" do
    sdk = Thesmsworks.test(S.jm(["entity", S.jm(["batch_message", S.jm([])])]))
    ent = Thesmsworks.batch_message(sdk)
    made = Thesmsworks.Entity.BatchMessage.create(ent, S.jm(["name", "test-create"]))
    assert S.ismap(made)
    assert S.getprop(made, "id") != nil
  end
end
